test/integration/short/control-connection-tests.js

```diff
@@ -92,16 +92,11 @@ describe('ControlConnection', function () {
       var cc = newInstance({ policies: { loadBalancing: new TestLoadBalancing() } });
       utils.series([
         cc.init.bind(cc),
-        function (next) {
-          //wait for all initial events
-          setTimeout(next, 5000);
-        },
-        function (next) {
-          helper.waitOnHost(function () {
-            //noinspection JSCheckFunctionSignatures
-            helper.ccmHelper.stopNode(2);
-          }, cc, 2, 'down', next);
-        },
+        // Don't stop the node until we know it's up.
+        helper.waitOnHostUp(cc, 2),
```

Comment:
One possible downside of this change is that we previously checked for Host 'down' / 'up' events to be emitted, instead we now check the state of the Host (isUp, or not present in the host map at all), which is set as a side effect of the status/topology events (like the Host events I suppose). Is this adequate?  I think it is, but can also add some logic to wait for the host events as well.